### ISSUE TITLE
do you mind give me a little help

### ISSUE THREAD

**Issue by hamwong** [eschava/broadlink-mqtt#2]
Hi, your code works perfectly well as I want, I really dislike to have an android for bridge, this is really a brilliant work. thank you.

I am a newbie on linux and raspberry pi, I got a stupid question to ask, is there anyway that I may run your code on background and auto start when system start up? so I don't have to manually activate your code.

Thanks and Regards

**Comment by eschava** [comment]
I use next way:

add to /etc/rc.local

# broadlink-mqtt
BROADLINKMQTTCONFIG=/home/cubie/broadlink-mqtt/mqtt.conf /home/cubie/broadlink-mqtt/mqtt.py &

where /home/cubie/broadlink-mqtt is path to broadlink-mqtt directory

**Comment by hamwong** [comment]
thanks for your promote response,
please excuse me as really leak on linux common knowledge,
do you mean I should add line in rc.local as below? the execute of mqtt.py should place at same line or second line?
BROADLINKMQTTCONFIG="/home/cubie/broadlink-mqtt/mqtt.conf"
/home/cubie/broadlink-mqtt/mqtt.py

Really appreciated for your help
Thanks and regards

**Comment by eschava** [comment]
all instruction should be in one line (except comment of course)
You can copy it from the link
https://gist.github.com/eschava/7698ca014ad6c323ccac08828bb25c7f

**Comment by hamwong** [comment]
Thank you, I see your reply instantly but I took a long while to google what does that mean.
I will do that now XD Thanks

**Comment by hamwong** [comment]
It's not work after reboot, I try to google with no result,
I wanna know does usually python require admin right? as I need to use sudo ./mqtt.py to start otherwise will have error and fail start, does it have any matter with it?

**Comment by eschava** [comment]
No, it doesn't need root privileges
Please check if your mqtt.py file has x attribute. Output of "ls -la mqtt.py" should be smth like
**-rwxr-xr-x** 1 cubie cubie 4327 Dec 21 19:47 mqtt.py

**Comment by hamwong** [comment]
you are right, as you told me it shouldn't require admin right, I look at the error msg in detail.

[19:02:33] pi@openHABianPi:/home/cubie/broadlink-mqtt$ ls -la mqtt.py
-rwxr-xr-x 1 pi pi 5252 Jan 30 02:30 mqtt.py
[19:02:47] pi@openHABianPi:/home/cubie/broadlink-mqtt$ ./mqtt.py
Traceback (most recent call last):
  File "./mqtt.py", line 13, in <module>
    logging.config.fileConfig('logging.conf')
  File "/usr/lib/python2.7/logging/config.py", line 85, in fileConfig
    handlers = _install_handlers(cp, formatters)
  File "/usr/lib/python2.7/logging/config.py", line 163, in _install_handlers
    h = klass(*args)
  File "/usr/lib/python2.7/logging/__init__.py", line 911, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/lib/python2.7/logging/__init__.py", line 936, in _open
    stream = open(self.baseFilename, self.mode)
IOError: [Errno 13] Permission denied: '/home/cubie/broadlink-mqtt/mqtt.log'

I think issue were come on mqtt.log, after I chmod 777 *.log, I able to perform mqtt.py by normal,
I will try restart when I back home (I am at office now), Thank you for all the help you provided, You are a nice gentleman XD

Thanks and Regards.

**Comment by hamwong** [comment]
I am really sorry to keep spam on  you, I found a wired thing, I don't know is this normal in linux,
I able to run mqtt.py in it's directory but no at root ,

[01:11:43] pi@openHABianPi:/$ BROADLINKMQTTCONFIG=/home/cubie/broadlink-mqtt/mqtt.conf /home/cubie/broadlink-mqtt/mqtt.py &
[1] 4035
[01:11:55] pi@openHABianPi:/$ Traceback (most recent call last):
  File "/home/cubie/broadlink-mqtt/mqtt.py", line 13, in <module>
    logging.config.fileConfig('logging.conf')
  File "/usr/lib/python2.7/logging/config.py", line 77, in fileConfig
    formatters = _create_formatters(cp)
  File "/usr/lib/python2.7/logging/config.py", line 113, in _create_formatters
    flist = cp.get("formatters", "keys")
  File "/usr/lib/python2.7/ConfigParser.py", line 607, in get
    raise NoSectionError(section)
ConfigParser.NoSectionError: No section: 'formatters'

Thanks and Regards

**Comment by eschava** [comment]
It's a bug. Fixed with new commit, please update
Now you can use just 
/home/cubie/broadlink-mqtt/mqtt.py &
to start app without BROADLINKMQTTCONFIG=...

**Comment by hamwong** [comment]
real thx for your super fast resposne. I will try it out now

**Comment by hamwong** [comment]
I got different error with below log, I currently solve it by add cd directory before and back to root after to avoid the error, do you think the issue is due to my linux incorrect setup on python or path$?

Traceback (most recent call last):
  File "/home/cubie/broadlink-mqtt/mqtt.py", line 14, in <module>
    logging.config.fileConfig(dirname + '/logging.conf')
  File "/usr/lib/python2.7/logging/config.py", line 85, in fileConfig
    handlers = _install_handlers(cp, formatters)
  File "/usr/lib/python2.7/logging/config.py", line 163, in _install_handlers
    h = klass(*args)
  File "/usr/lib/python2.7/logging/__init__.py", line 911, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/lib/python2.7/logging/__init__.py", line 936, in _open
    stream = open(self.baseFilename, self.mode)
IOError: [Errno 13] Permission denied: '/mqtt.log'

**Comment by hamwong** [comment]
after update to new version, keep discount and connect, even I disconnect power of RM Pro,
it still keep connect and disconnect
[2017-02-01 02:20:07,047] DEBUG OOOOPS! Broadlink disconnects
[2017-02-01 02:20:18,059] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#
[2017-02-01 02:20:29,074] DEBUG OOOOPS! Broadlink disconnects
[2017-02-01 02:20:40,088] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#
[2017-02-01 02:20:51,103] DEBUG OOOOPS! Broadlink disconnects
[2017-02-01 02:21:02,118] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#
[2017-02-01 02:21:13,128] DEBUG OOOOPS! Broadlink disconnects
[2017-02-01 02:21:24,142] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#
[2017-02-01 02:21:35,148] DEBUG OOOOPS! Broadlink disconnects
[2017-02-01 02:21:46,163] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#
[2017-02-01 02:21:57,178] DEBUG OOOOPS! Broadlink disconnects
[2017-02-01 02:22:08,192] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#
[2017-02-01 02:22:19,207] DEBUG OOOOPS! Broadlink disconnects

**Comment by eschava** [comment]
> IOError: [Errno 13] Permission denied: '/mqtt.log'

Fixed with new commit

> [2017-02-01 02:20:07,047] DEBUG OOOOPS! Broadlink disconnects
> [2017-02-01 02:20:18,059] DEBUG Connected to MQTT broker, subscribing to topic broadlink/#

Could be related to two instances of mqtt.py run at the same time

**Comment by hamwong** [comment]
it's work now with your new changes on code, I able to active mqtt at root
regarding two instances of mqtt.py issue, I don't know how does it happen as I just got restart to try your new code, but after I shutdown the raspberry and on again, issue solved so I deleted the comment.

as I use ssh to raspberry, I have stop put in rc.local as it will have issue as  more than 1 terminal operate will have multi instances, I usually open few terminal for difference purpose. Anyway, Thank you in deep to all the way to help a newbie. Thank you again.
